source/module_hamilt/test/nonlocal_test.cpp

```diff
@@ -0,0 +1,116 @@
+    };
+
+    std::vector<std::complex<double>> expected_ps = {
+      std::complex<double>(1.08811,0), std::complex<double>(0,2.11482e-17), std::complex<double>(0,4.22963e-17), std::complex<double>(0,-6.34445e-17), std::complex<double>(0.473301,2.11482e-17), std::complex<double>(-3.80667e-16,-5.28704e-18), std::complex<double>(-2.11482e-17,-2.11482e-17), std::complex<double>(4.22963e-17,2.64352e-17), std::complex<double>(1.50743e-33,1.75712e-17), std::complex<double>(0.93285,-5.71206e-17), std::complex<double>(0,-2.83751e-33), std::complex<double>(-3.83371e-17,1.41876e-33), 
+      std::complex<double>(6.38951e-18,2.79541e-18), std::complex<double>(-0.0996506,1.91685e-17), std::complex<double>(1.2779e-17,5.11161e-17), std::complex<double>(2.87528e-17,3.83371e-17), std::complex<double>(-2.48283e-33,7.02847e-17), std::complex<double>(1.2779e-17,-2.83751e-33), std::complex<double>(0.93285,-5.71206e-17), std::complex<double>(-1.02232e-16,3.54689e-34), std::complex<double>(0,6.38951e-18), std::complex<double>(9.58427e-18,5.11161e-17), std::complex<double>(-0.0996506,-6.38951e-18), 
+      std::complex<double>(-7.83714e-18,0), std::complex<double>(1.41876e-33,1.24596e-16), std::complex<double>(-5.11161e-17,1.41876e-33), std::complex<double>(-1.91685e-17,3.54689e-34), std::complex<double>(0.93285,-5.71206e-17), std::complex<double>(0,5.09164e-18), std::complex<double>(2.87528e-17,2.55581e-17), std::complex<double>(-7.78722e-18,0), std::complex<double>(-0.0996506,6.38951e-18), std::complex<double>(0.473301,-2.11482e-17), std::complex<double>(3.80667e-16,-5.28704e-18), 
+      std::complex<double>(2.11482e-17,-2.11482e-17), std::complex<double>(-4.22963e-17,2.64352e-17), std::complex<double>(1.08811,-6.92094e-34), std::complex<double>(-2.95657e-33,-2.11482e-17), std::complex<double>(-1.31476e-33,4.22963e-17), std::complex<double>(3.35836e-34,1.26889e-16), std::complex<double>(-6.38951e-18,-3.99345e-18), std::complex<double>(-0.0996506,6.38951e-18), std::complex<double>(1.2779e-17,-4.47266e-17), std::complex<double>(2.87528e-17,-4.47266e-17), 
+      std::complex<double>(-5.32034e-34,-6.38951e-18), std::complex<double>(0.93285,-5.71206e-17), std::complex<double>(-1.2779e-17,1.41876e-33), std::complex<double>(2.55581e-17,-5.67503e-33), std::complex<double>(0,-6.38951e-18), std::complex<double>(1.2779e-17,-4.47266e-17), std::complex<double>(-0.0996506,1.91685e-17), std::complex<double>(9.98362e-19,0), std::complex<double>(1.41876e-33,-1.91685e-17), std::complex<double>(0,-1.41876e-33), std::complex<double>(0.93285,-5.71206e-17), 
+      std::complex<double>(1.02232e-16,7.80316e-33), std::complex<double>(-6.38951e-18,-3.5941e-18), std::complex<double>(2.87528e-17,-1.2779e-17), std::complex<double>(-5.34123e-18,0), std::complex<double>(-0.0996506,0), std::complex<double>(0,-3.51423e-17), std::complex<double>(1.2779e-17,-4.96565e-33), std::complex<double>(8.30637e-17,-2.12814e-33), std::complex<double>(0.93285,-5.71206e-17)
```

Comment:
How about replacing these stuff with `std::vector<std::complex<double>>  v={{1,0},{2,1}};`?

Reply:
In this case, the imag part of the complex number cannot be recognized during my test process.

Reply:
> In this case, the imag part of the complex number cannot be recognized during my test process.

That's weird, this works fine on my compiler with `std=c++11`.

Reply:
It has been solved. I misused the initialization symbol of complex.